RocketChat/Rocket.Chat
# fix: Agents can't leave omnichannel rooms that have already been closed

## Proposed changes (including videos or screenshots)

 - Allow users to leave omnichannel rooms after they have been closed (by the visitor). If a user tries to close a room that has already been closed AND they have a subscription in this room, we'll remove them from the room instead of throwing an error;
 - Display button to close livechat rooms when the user is still subscribed to the room and has any of the permissions (`close-livechat-room` or `close-others-livechat-room`) to close a room.

Due to some **race condition**, users could end up joining omnichannel rooms that have already been closed, which caused them to get stuck in there.

## Issue(s)


## Steps to test or reproduce

Since having a user stuck in an omnichannel room after it's been closed depends on a race condition, reproducing this issue may require manually inserting a subscription in the DB.
With a user that is stuck in an omnichannel/livechat room, try to close the room again through the UI or using the `livechat.closeRoomByUser` endpoint OR the `livechat:closeRoom` method. Both options currently throw an error in this case, but with this PR, the subscription is removed and a succesful result should be returned.

## Further comments

I decided not to add this change to the `livechat.closeRoom` endpoint since this one is meant to be used by visitors -- which are not subscribed to the room directly, only agents are.

[SUP-612](https://rocketchat.atlassian.net/browse/SUP-612)


[SUP-612]: https://rocketchat.atlassian.net/browse/SUP-612?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
